test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTypesTest.java

```diff
@@ -0,0 +1,1215 @@
+                                         : row(0, tuple(1, 2), tuple(3, 4), null))) // TODO: is this correct?
+                .deleteColumn("DELETE a.x FROM %s WHERE k=0",
+                              "b.y",
+                              rows(row(4)),
+                              rows(row(0, tuple(null, 2), tuple(3, 4), 10)),
+                              rows(row(0, tuple(null, 2), tuple(3, 4), 10))) // unselected 'a' is repaired, see CASSANDRA-10657
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, tuple(null, 2), tuple(3, 4), 10)))
+                .tearDown();
+    }
+
+    private Tester tester()
+    {
+        return new Tester(strategy, flush, localCoordinator, paging);
+    }
+
+    private static class Tester
+    {
+        private static final AtomicInteger seqNumber = new AtomicInteger();
+        private static final Object[][] EMPTY = new Object[0][];
+
+        private final String tableName = KEYSPACE + ".t_" + seqNumber.getAndIncrement();
+        private final ReadRepairStrategy strategy;
+        private final boolean flush;
+        private final boolean paging;
+        private final ICoordinator coordinator;
+
+        private String restriction; // the tested CQL query WHERE restriction
+        private String allColumnsQuery; // a SELECT * query for the table using the tested restriction
+
+        Tester(ReadRepairStrategy strategy, boolean flush, boolean localCoordinator, boolean paging)
+        {
+            this.strategy = strategy;
+            this.flush = flush;
+            this.paging = paging;
+
+            // decide which node we are going to use as coordinator for distributed reads, either the one containing the
+            // fresh data or the outdated one
+            coordinator = cluster.coordinator(localCoordinator ? 1 : 2);
+        }
+
+        Tester schemaChange(String... queries)
+        {
+            for (String query : queries)
+                cluster.schemaChange(withKeyspace(query));
+            return this;
+        }
+
+        Tester createTable(String createTable)
+        {
+            return schemaChange(String.format(createTable + " WITH read_repair='%s'", tableName, strategy));
+        }
+
+        /**
+         * Runs the specified insertions in only one replica.
+         */
+        Tester populateNode(String... queries)
+        {
+            // run the write queries only on one node
+            for (String query : queries)
+                cluster.get(1).executeInternal(String.format(query, tableName));
+
+            // flush the update node to ensure reads come from sstables
+            if (flush)
+                cluster.get(1).flush(KEYSPACE);
+
+            return this;
+        }
+
+        /**
+         * Sets the {@code WHERE...} CQL query restriction to be tested.
+         */
+        Tester withQueryRestriction(String restriction)
```

Comment:
nit: Can we just pass the restriction to the constructor and make it `final` if we're not actually going to reuse `Tester` instances?